app/helpers/logging.py

```diff
@@ -51,11 +51,8 @@ def tick(self):
         if self.verbosity >= 5:
```

Comment:
Possibility to merge this part of the code also, but I think code would be less clear
```suggestion
        sould_log = self.verbosity >= 5 or \
                    self.current_step % max(1, int(math.pow(10, (5 - self.verbosity)))) == 0 or \
                    self.current_step == self.total_steps
```